go/vt/vtgr/vtgr.go

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright 2021 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vtgr
+
+import (
+	"flag"
+	"strings"
+	"sync"
+	"time"
+
+	"vitess.io/vitess/go/vt/vtgr/config"
+
+	"vitess.io/vitess/go/vt/vtgr/db"
+
+	"golang.org/x/net/context"
+
+	"vitess.io/vitess/go/vt/log"
+	"vitess.io/vitess/go/vt/topo"
+	"vitess.io/vitess/go/vt/vtgr/controller"
+	"vitess.io/vitess/go/vt/vttablet/tmclient"
+)
+
+var (
+	refreshInterval      = flag.Duration("refresh_interval", 10*time.Second, "refresh interval to load tablets")
+	scanInterval         = flag.Duration("scan_interval", 3*time.Second, "scan interval to diagnose and repair")
+	scanAndRepairTimeout = flag.Duration("scan_repair_timeout", 3*time.Second, "time to wait for a Diagnose and repair operation")
+	vtgrConfigFile       = flag.String("vtgr_config", "", "config file for vtgr")
+
+	localDbPort = flag.Int("db_port", 0, "local mysql port, set this to enable local fast check")
+)
+
+// VTGR manages VTGR
```

Comment:
A more descriptive comment will be good.

Reply:
Added more comment